src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -367,12 +355,15 @@ public int Available
                         _nonBlockingConnectInProgress = false;
                     }
 
-                    if (_rightEndPoint == null)
+                    if (_rightEndPoint == null || !_isConnected)
                     {
                         return null;
                     }
 
-                    Internals.SocketAddress socketAddress = IPEndPointExtensions.Serialize(_rightEndPoint);
+                    Internals.SocketAddress socketAddress =
+                        _addressFamily == AddressFamily.InterNetwork || _addressFamily == AddressFamily.InterNetworkV6 ?
+                            IPEndPointExtensions.Serialize(_rightEndPoint) :
+                            new Internals.SocketAddress(_addressFamily, SocketPal.MaximumAddressSize); // may be different size thaan _rightEndPoint.
```

Comment:
The comment has a typo (`thaan`).
Should we make `IPEndPointExtensions` throw when given a non `IPEndPoint` type?
Or maybe rename it to `EndPointExtensions` and let it handle other types too, like the unix address?